File: day01/puzzle.js
import { createAocPuzzle } from "./aoc.js";

const testAnswers = {
  part1: 7,
  part2: 5
};

function parseInput(input) {
  return input.map(line => +line);
}

function runPart1(input) {
  let result = 0;
  for (let i = 1; i < input.length; ++i) {
    if (input[i] > input[i - 1]) {
      ++result;
    }
  }
  return result;
}

function runPart2(input) {
  let result = 0;
  for (let i = 1, len = input.length - 2; i < len; ++i) {
    const common = input[i] + input[i + 1];
    if (common + input[i + 2] > input[i - 1] + common) {
      ++result;
    }
  }
  return result;
}

export function createPuzzle() {
  return createAocPuzzle({ parseInput, runPart1, runPart2, testAnswers });
}

File: testrun/puzzle.py
from aoc import AocPuzzle


TEST_ANSWERS = {
    "part1": 2421,
    "part2": 335
}


def is_prime(number: int):
    if number == 2 or number == 3:
        return True
    elif number < 2 or number % 2 == 0 or number % 3 == 0:
        return False

    factor = 5
    while factor * factor <= number:
        if number % factor == 0 or number % (factor + 2) == 0:
            return False

        factor += 6

    return True


class Puzzle(AocPuzzle[int, int]):
    def __init__(self):
        super().__init__(TEST_ANSWERS)


    def parse_input(self, lines: list[str]):
        return [int(line) for line in lines]


    def run_part1(self):
        result = 0
        for i, number in enumerate(self._input):
            if is_prime(number):
                result += i * number

        return result


    def run_part2(self):
        result = 0
        for i, number in enumerate(self._input):
            if not is_prime(number):
                if i % 2 == 0:
                    result += number
                else:
                    result -= number

        return result
